packages/core/src/resources/internal-registry.ts

```diff
@@ -0,0 +1,426 @@
+import { ResourceMetadata } from './types.js';
```

Comment:
this file name is confusing

if we do decide to continue with the `internal` naming, then:

make a separate folder for internal resource management: `resources/internal`

to this folder:

- add types.ts which defines the interface shapes/types (currently all handled in internal-registry.ts)

- add filehandler.ts which defines the `FileSystemResourceHandler`

- add `provider.ts` that aggregates the different handlers (current internal-provider.ts) 

this is also more like our tools modules which follows the above structure for internal tools

this better separates the layers -> manager.ts is top level aggregator
internal -> handles internal resources